### PR TITLE
NFT reconfigureFundingCyclesOf tx

### PR DESCRIPTION
**Pull request by johnnyd-eth**
## What does this PR do and why?

- Allows projects without NFTs to deploy to have NFTs:
      1. Call `setOperator` tx
      2. Call `deployer.reconfigureFundingCyclesOf` tx

Screen Shot 2022-11-04 at 12 42 17 pm

Screen Shot 2022-11-04 at 12 42 27 pm

(note: screenshots missing the NFT rewards in the review section)
      
- Adds NFT rewards to summary section of the reconfig review (this is very ugly, I know. I think we should just wait to redo the whole thing to a similar style as the new FC card in a follow-up)

- Also note: as we discussed @tomquirk the reconfigFC page has a bug that resets the redux state after some period of time. This is more likely to happen in this case because of the wait for the `setOperator` tx. We should definitely try to get it fixed in a follow-up before mainnet release. 

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [ ] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

File: src/components/v2v3/shared/FundingCycleConfigurationDrawers/NftDrawer/NftRewardTierCard.tsx [hunk 1]
```diff
@@ -101,14 +102,15 @@ export default function NftRewardTierCard({
...
+          {rewardTier.maxSupply &&
+          rewardTier.maxSupply !== DEFAULT_NFT_MAX_SUPPLY ? (
             <div style={{ fontSize: 13, marginTop: 15 }}>
               <Trans>
                 <strong>Max. supply:</strong>{' '}
                 <span>{rewardTier.maxSupply}</span>
               </Trans>
             </div>
-          )}
+          ) : null}
```
Comment (johnnyd-eth) [inline src/components/v2v3/shared/FundingCycleConfigurationDrawers/NftDrawer/NftRewardTierCard.tsx:113]: Unrelated fix

File: src/components/v2v3/V2V3Project/V2V3ProjectSettings/pages/ReconfigureFundingCycleSettingsPage/ReconfigureFundingCycleForm.tsx [hunk 2]
```diff
@@ -155,18 +163,35 @@ export function V2V3ReconfigureFundingCycleForm() {
...
-            <Trans>Deploy funding cycle configuration</Trans>
-          </span>
-        </Button>
+        {nftDrawerHasSavedChanges ? (
+          <Space size="middle" direction="vertical">
+            <div style={{ display: 'flex' }}>
+              <h2 style={{ marginRight: 5 }}>1.</h2>
+              <SetNftOperatorPermissionsButton
+                onConfirmed={() => setNftOperatorConfirmed(true)}
+              />
```
Comment (tomquirk) [inline src/components/v2v3/V2V3Project/V2V3ProjectSettings/pages/ReconfigureFundingCycleSettingsPage/ReconfigureFundingCycleForm.tsx:174]: **suggestion** perhaps hide or disable this if the permission has already been granted?

File: src/components/v2v3/V2V3Project/V2V3ProjectSettings/pages/ReconfigureFundingCycleSettingsPage/SetNftOperatorPermissionsButton.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+          fontSize: '0.7rem',
+        }}
+      >
+        Allow the{' '}
+        <ExternalLink
+          href={`https://github.com/jbx-protocol/juice-721-delegate/blob/main/contracts/JBTiered721DelegateDeployer.sol`}
+        >
+          Juicebox NFT deployer contract
+        </ExternalLink>{' '}
+        to executed reconfiguration transactions on this project's behalf.
```
Comment (tomquirk) [inline src/components/v2v3/V2V3Project/V2V3ProjectSettings/pages/ReconfigureFundingCycleSettingsPage/SetNftOperatorPermissionsButton.tsx:60]:
```suggestion
        to reconfigure this project's funding cycle.
```

File: src/hooks/v2v3/transactor/SetNftOperatorPermissionsTx.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+import {
+  handleTransactionException,
+  TransactorInstance,
+} from 'hooks/Transactor'
+import { useWallet } from 'hooks/Wallet'
+import { V2OperatorPermission } from 'models/v2v3/permissions'
+import { useContext } from 'react'
+import invariant from 'tiny-invariant'
+
+export function useSetNftOperatorPermissionsTx(): TransactorInstance {
```
Comment (tomquirk) [inline src/hooks/v2v3/transactor/SetNftOperatorPermissionsTx.tsx:15]: **suggestion** can we at least make this a generic `useSetOperator` that accepts the relevent args? Then pass the nft args in?

Comment (tomquirk) [reply src/hooks/v2v3/transactor/SetNftOperatorPermissionsTx.tsx:15]: actually don't worry about this. Non-blocking, can fix later.

File: src/components/v2v3/V2V3Project/V2V3ProjectSettings/pages/ReconfigureFundingCycleSettingsPage/ReconfigureFundingCycleForm.tsx (hunk 2 above)
Comment (johnnyd-eth) [reply src/components/v2v3/V2V3Project/V2V3ProjectSettings/pages/ReconfigureFundingCycleSettingsPage/ReconfigureFundingCycleForm.tsx:174]: Done this now. Tested and it's working well 👍

File: src/components/v2v3/V2V3Project/V2V3ProjectSettings/pages/ReconfigureFundingCycleSettingsPage/ReconfigureFundingCycleForm.tsx [hunk 5]
```diff
@@ -155,18 +183,35 @@ export function V2V3ReconfigureFundingCycleForm() {
...
-          loading={reconfigureLoading}
-          onClick={reconfigureFundingCycle}
-          disabled={!fundingHasSavedChanges && !nftsWithFalseDataSourceForPay}
-          type="primary"
-        >
-          <span>
-            <Trans>Deploy funding cycle configuration</Trans>
-          </span>
-        </Button>
+        {nftDrawerHasSavedChanges && !nftContractHasPermission ? (
```
Comment (tomquirk) [inline src/components/v2v3/V2V3Project/V2V3ProjectSettings/pages/ReconfigureFundingCycleSettingsPage/ReconfigureFundingCycleForm.tsx:188]: **suggestion** can we put this in its own file?

Comment (tomquirk) [reply src/components/v2v3/V2V3Project/V2V3ProjectSettings/pages/ReconfigureFundingCycleSettingsPage/ReconfigureFundingCycleForm.tsx:188]: eh. watevs. leave for now

**Review by tomquirk — APPROVED** [review]
